index/src/executors/bridge_executors/phala_to_evm.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{
+    prelude::{Address, Executor},
+    subrpc::{create_transaction, send_transaction},
+    traits::{Amount, Error},
+};
+use alloc::string::{String, ToString};
+use xcm::v1::MultiAsset;
+use xcm::v1::{AssetId, Fungibility, Junction, Junctions, MultiLocation};
+
+pub struct Phala2EvmExecutor {
+    rpc: String,
+}
+
+impl Executor for Phala2EvmExecutor {
+    fn new(
+        _bridge_address: crate::traits::Address,
+        _abi_json: &[u8],
+        rpc: &str,
+    ) -> core::result::Result<Self, crate::traits::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Self {
+            rpc: rpc.to_string(),
+        })
+    }
+
+    fn transfer(
+        &self,
+        signer: [u8; 32],
+        _token_rid: primitive_types::H256, // TODO: this param is useless,  that means the interface need to be changed
```

Comment:
Yes, I suggest the bridge-specific data can be sealed inside the executor, as I suggest in [sketch branch](https://github.com/Phala-Network/index-contract/blob/110fb512b1ce1ea1e3fa3742fe57fcf296bb4b95/index/src/executors/bridge_executor.rs#L17)

Reply:
We can leave it to later PR